google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -174,6 +174,92 @@ void WriteObject(google::cloud::storage::Client client, int& argc,
   (std::move(client), bucket_name, object_name, desired_line_count);
 }
 
+void GenerateEncryptionKey(google::cloud::storage::Client client, int& argc,
+                           char* argv[]) {
+  if (argc != 1) {
+    throw Usage{"generate-encryption-key"};
+  }
+  //! [generate encryption key] [START generate_encryption_key_base64]
+  // Create a pseudo-random number generator (PRNG), this is included for
+  // demonstration purposes only. You should consult your security team about
+  // best practices to initialize PRNG. In particular, you should verify that
+  // the C++ library and operating system provide enough entropy to meet the
+  // security policies in your organization.
+
+  // Use the Mersenne-Twister Engine in this example:
+  //   https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine
+  // Any C++ PRNG can be used below, the choice is arbitrary.
+  using GeneratorType = std::mt19937_64;
+
+  // Create the default random device to fetch entropy.
+  std::random_device rd;
+
+  // Compute how much entropy we need to initialize the GeneratorType:
+  constexpr auto kRequiredEntropyWords =
+      GeneratorType::state_size *
+      (GeneratorType::word_size / std::numeric_limits<unsigned int>::digits);
+
+  // Capture the entropy bits into a vector.
+  std::vector<unsigned long> entropy(kRequiredEntropyWords);
+  std::generate(entropy.begin(), entropy.end(), [&rd] { return rd(); });
+
+  // Create the PRNG with the fetched entropy.
+  std::seed_seq seed(entropy.begin(), entropy.end());
+  GeneratorType gen(seed);
+
+  namespace gcs = google::cloud::storage;
+  gcs::EncryptionKeyData data = gcs::CreateKeyFromGenerator(gen);
```

Comment:
I appreciate the example of using a PRNG to generate a seed.

At first it was a little overwhelming compared to other examples on the site.

The example could be shortened by removing the bit on entropy.

Reply:
Default initialized PRNGs in C++ are completely predictable, we should be careful not to give poor (and dangerous) examples to our users.  What do you think of removing all the entropy initialization stuff, and add a link to our code or a gist that properly initializes the generator?

Reply:
Thanks Carlos. If it's predictable then the case, I'd rather lean towards giving developers direction than a subtle example with missing pieces. Please keep as-is.

Reply:
Ack. Done.